In [1]:
import sys
#Indicate operating environment and import core modules
location_input = input("what computer are you on? a = Bens, b = gpucluster, c = other   ")
location_dict = {'a': "C:\\Users\\heine\\github\\expression_broad_data", 'b': "/home/heineike/github/expression_broad_data",'c':'you need to add your location to the location_dict'}
base_dir = location_dict[location_input]
print("base directory is " + base_dir)

if sys.path[-1] != base_dir:
    sys.path.append(base_dir)
    print("Added " + base_dir + " to path: " )
    print(sys.path)
import os

print("I am about to import a library")
from core import expression_plots 
from core import io_library 
%load_ext autoreload
%autoreload 2
# %matplotlib 
# plt.ioff()
#%matplotlib inline
%matplotlib notebook
data_processing_dir = base_dir + os.sep + os.path.normpath("expression_data") + os.sep

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns; sns.set(style="ticks", color_codes=True)
from sklearn import linear_model
import pickle
import subprocess
import networkx as nx

from Bio import SeqIO
from Bio import SeqFeature as sf
from Bio.Alphabet import generic_dna
from Bio.Seq import Seq

import re

# from collections import Counter
# import scipy.stats as stats
# from itertools import chain
#this only works if you are online
online_input = input("are you online? Yes/No")
if online_input == "Yes": 
    import plotly.plotly as py
    import plotly.graph_objs as pygo
    import plotly.tools as pytools
    py.sign_in('heineike02', 'APPjKrtARaN2ZgUYIkqr')

what computer are you on? a = Bens, b = gpucluster, c = other   b
base directory is /home/heineike/github/expression_broad_data
Added /home/heineike/github/expression_broad_data to path: 
['', '/usr/lib/python35.zip', '/usr/lib/python3.5', '/usr/lib/python3.5/plat-x86_64-linux-gnu', '/usr/lib/python3.5/lib-dynload', '/usr/local/lib/python3.5/dist-packages', '/usr/local/lib/python3.5/dist-packages/cmonkey2-1.2.6-py3.5.egg', '/usr/local/lib/python3.5/dist-packages/SQLAlchemy_Utils-0.32.14-py3.5.egg', '/usr/local/lib/python3.5/dist-packages/svgwrite-1.1.11-py3.5.egg', '/usr/local/lib/python3.5/dist-packages/Routes-2.4.1-py3.5.egg', '/usr/local/lib/python3.5/dist-packages/CherryPy-10.2.2-py3.5.egg', '/usr/local/lib/python3.5/dist-packages/biopython-1.69-py3.5-linux-x86_64.egg', '/usr/local/lib/python3.5/dist-packages/configparser-3.5.0-py3.5.egg', '/usr/local/lib/python3.5/dist-packages/rpy2-2.8.6-py3.5-linux-x86_64.egg', '/usr/local/lib/python3.5/dist-packages/repoze.lru-0.6-py3.5.egg', '

In [2]:
%load_ext autoreload
%autoreload 2
# %matplotlib 
# plt.ioff()
#%matplotlib inline
%matplotlib notebook

The autoreload extension is already loaded. To reload it, use:
  %reload_ext autoreload


# Identify all paralogs
subset paralogs by whole genome duplication. 

In [3]:
#Total number of WGD paralogs
ohnologs = pd.read_csv(data_processing_dir + os.path.normpath("ortholog_files_YGOB/ohnologs.csv"))
print('Total number of Ohnologs: ' + str(len(ohnologs)))

Total number of Ohnologs: 547


In [4]:
#Total number of KL genes that have a paralog in SCER
fname_cross_spec_comparison = data_processing_dir + os.path.normpath("20170817_klscpka/cross_species_compare_ASminus_ASplus_YPD.csv")
kl_sc_PKA_data = pd.read_csv(fname_cross_spec_comparison, index_col=0)

kl_sc_PKA_data_paralogs = kl_sc_PKA_data[kl_sc_PKA_data["N SCer Orthologs"]==2]

print('Total number of KL genes that map to a set of paralogs in S.Cer: ' + str(int(len(kl_sc_PKA_data_paralogs)/2)))

#Total of these that are WGD paralogs
ohnolog_set = set(ohnologs['Gene 1']) | set(ohnologs['Gene 2'])

non_ohnologs = set(kl_sc_PKA_data_paralogs['sc_genename']) - ohnolog_set
ohnologs_klac_orth = set(kl_sc_PKA_data_paralogs['sc_genename']) & ohnolog_set

print('KL genes that map to a set of Ohnologs: ' + str(int(len(ohnologs_klac_orth)/2)) + '\n' +
     'KL genes that map to a non Ohnolog paralog: ' + str(int(len(non_ohnologs))))


Total number of KL genes that map to a set of paralogs in S.Cer: 519
KL genes that map to a set of Ohnologs: 519
KL genes that map to a non Ohnolog paralog: 0


Apparently the ortholog mapping I am using only includes paralogs from the WGH.  I am sure there must be others.  I should probably investigate this at some stage. 

In [120]:
#What happens under PKA inhibition for SCer Genes that don't have a K.Lactis ortholog? 

ohnologs_no_kl_orth = ohnolog_set-set(kl_sc_PKA_data_paralogs['sc_genename'])

#Doesn't work - pickle breaks between gpu and my computer
#YPD data
# fname_comparison_data = "20170817_klscpka/ASminus_ASplus_YPD_comparison.pkl"
# with open(data_processing_dir + os.path.normpath(fname_comparison_data), 'rb') as pkl_in:
#      comparison_data = pickle.load(pkl_in)

sc_ypd_foldchange = pd.read_csv(data_processing_dir + os.path.normpath('20170817_klscpka/20170817_klscpka_DESEQ_YPD_t40_AsMinvAsPlus_SC.csv'))
sc_avg_counts = pd.read_csv(data_processing_dir + os.path.normpath('20170817_klscpka/20170817_klscpka_avg_counts_SC.csv'))
sc_avg_counts.rename(columns={'0':'sc_genename'},inplace=True)

sc_ypd_data = sc_ypd_foldchange.rename(columns= {'Unnamed: 0':'sc_genename'})
sc_ypd_data.set_index('sc_genename', inplace = True)
sc_ypd_data = sc_ypd_data.assign(SC_common_name = io_library.SC_common_name_lookup(sc_ypd_data.index))

sc_ypd_data = sc_avg_counts.join(sc_ypd_data, on='sc_genename')
sc_ypd_data.set_index('sc_genename', inplace = True)
sc_ypd_data


AS_noNMPP1_SDC_t40_mean  AS_noNMPP1_SDC_t40_var  \
sc_genename                                                    
YAL012W                     3.413158                0.014747   
YAL069W                     0.000000                0.000000   
YAL068W-A                   0.000000                0.000000   
YAL068C                     0.000000                0.000000   
YAL067W-A                   0.000000                0.000000   
YAL067C                     0.352827                0.098438   
YAL066W                     0.000000                0.000000   
YAL065C                     0.193503                0.112330   
YAL064W-B                   0.000000                0.000000   
YAL064C-A                   0.000000                0.000000   
YAL064W                     0.227815                0.041943   
YAL063C-A                   0.284893                0.061814   
YAL063C                     1.411147                0.007389   
YAL062W                     1.724973                0.017682   
YAL061W                     1.351845                0.019829   
YAL060W                     2.753511                0.002882   
YAL059C-A                   0.145445                0.063463   
YAL059W                     2.524395                0.053120   
YAL058W                     2.015586                0.003435   
YAL056C-A                   0.000000                0.000000   
YAL056W                     2.854256                0.011523   
YAL055W                     1.790064                0.027135   
YAL054C                     0.734570                0.098092   
YAL053W                     2.603907                0.011013   
YAL051W                     1.760981                0.009733   
YAL049C                     2.205739                0.155863   
YAL048C                     0.441193                0.024616   
YAL047W-A                   0.441193                0.024616   
YAL047C                     1.348607                0.002629   
YAL046C                     2.707204                0.014094   
...                              ...                     ...   
YMR303C                     0.755335                0.013239   
YMR304W                     2.085581                0.022228   
YMR304C-A                   2.544195                0.005735   
YMR305C                     2.150329                0.236797   
YMR306W                     0.925550                0.059479   
YMR306C-A                   0.095594                0.027415   
YMR307W                     3.213080                0.044754   
YMR307C-A                   0.000000                0.000000   
YMR308C                     2.842319                0.007838   
YMR309C                     1.908942                0.008289   
YMR310C                     2.044595                0.010592   
YMR311C                     2.455953                0.014764   
YMR312W                     2.020864                0.004937   
YMR313C                     2.082615                0.003827   
YMR314W                     2.842132                0.004751   
YMR315W                     2.877316                0.007157   
YMR315W-A                   0.000000                0.000000   
YMR316W                     2.360716                0.027278   
YMR316C-A                   0.441193                0.024616   
YMR316C-B                   2.115821                0.002468   
YMR317W                     0.726196                0.031322   
YMR318C                     3.484530                0.003403   
YMR319C                     1.643120                0.056432   
YMR320W                     0.193503                0.112330   
YMR321C                     0.000000                0.000000   
YMR322C                     0.000000                0.000000   
YMR323W                     0.000000                0.000000   
YMR324C                     0.000000                0.000000   
YMR325W                     0.000000                0.000000   
YMR326C                     0.000000         

In [144]:
sc_ypd_data[sc_ypd_data['SC_common_name'].isin(['MSN2','MSN4'])][['AS_noNMPP1_YPD_t40_mean','AS_yesNMPP1_YPD_t40_mean']]

AS_noNMPP1_YPD_t40_mean  AS_yesNMPP1_YPD_t40_mean
sc_genename                                                   
YKL062W                     0.932825                  1.448539
YMR037C                     0.878076                  1.113982

In [121]:
ohnologs_no_kl_orth_pairs = ohnologs[ohnologs['Gene 1'].isin(ohnologs_no_kl_orth)]

print('Pairs of paralogs in SC with no kl ortholog: ' + str(len(ohnologs_no_kl_orth_pairs)))

#Go through this and assign fold change to each one.  

new_data = {'gene1_LFC': [], 'gene1_avg_exp_noNMPP1': [],'gene1_avg_exp_yesNMPP1': [],
            'gene2_LFC': [], 'gene2_avg_exp_noNMPP1': [],'gene2_avg_exp_yesNMPP1': []}

for ind, row in ohnologs_no_kl_orth_pairs.iterrows(): 
    gene1 = row['Gene 1']
    new_data['gene1_LFC'].append(float(sc_ypd_data.loc[gene1]['log2FoldChange']))
    new_data['gene1_avg_exp_noNMPP1'].append(float(sc_ypd_data.loc[gene1]['AS_noNMPP1_YPD_t40_mean']))
    new_data['gene1_avg_exp_yesNMPP1'].append(float(sc_ypd_data.loc[gene1]['AS_yesNMPP1_YPD_t40_mean']))
    
    gene2 = row['Gene 2']
    new_data['gene2_LFC'].append(float(sc_ypd_data.loc[gene2]['log2FoldChange']))
    new_data['gene2_avg_exp_noNMPP1'].append(float(sc_ypd_data.loc[gene2]['AS_noNMPP1_YPD_t40_mean']))
    new_data['gene2_avg_exp_yesNMPP1'].append(float(sc_ypd_data.loc[gene2]['AS_yesNMPP1_YPD_t40_mean']))

ohnologs_no_kl_orth_pairs.assign(**new_data)[['Gene Name 1','gene1_LFC','gene1_avg_exp_noNMPP1','gene1_avg_exp_yesNMPP1','Gene Name 2', 'gene2_LFC','gene2_avg_exp_noNMPP1','gene2_avg_exp_yesNMPP1']].sort_values(by='gene1_LFC')

# for ind, column in enumerate(columns): 
#     a[column] = [row[ind] for row in b]

Pairs of paralogs in SC with no kl ortholog: 28


Gene Name 1  gene1_LFC  gene1_avg_exp_noNMPP1  gene1_avg_exp_yesNMPP1  \
520       SPO19  -4.840507               0.713965                0.000000   
534      RPL20A  -3.936965               3.412258                2.367661   
543        ATR1  -2.128023               2.374841                1.862896   
435      RPL41A  -2.036833               3.657479                3.178033   
417        TIR3  -1.511669               2.460936                2.132716   
215        MIG3  -1.339809               1.399376                1.173717   
190        BAP2  -1.192563               2.766405                2.572179   
175     YDL012C  -0.955096               2.531284                2.382145   
293        RSC3  -0.903158               1.856831                1.758987   
31         LCB3  -0.753349               2.341170                2.248597   
65         PMP1  -0.629034               1.480049                1.473213   
247       ENV11  -0.234235               1.470594                1.543491   
397        FIG2  -0.132859               2.052604                2.118499   
348        PDR8   0.169915               1.817600                2.000822   
416       PDR11   0.651212               1.324514                1.624042   
187        YRO2   0.693128               1.303044                1.775621   
300        HXT6   1.151610               0.502415                0.930997   
306        YPR1   1.316042               2.395377                2.926154   
330       IRC23   1.364321               1.411082                1.970659   
279        RTA1   1.444121               1.028044                1.583572   
240     YGR035C   1.680295               0.522243                1.032656   
153        HYR1   1.965184               3.313531                4.042196   
537        HOR7   2.073966               3.626752                4.357399   
310        PDR5   2.312915               2.239221                3.056676   
45      YBR284W   3.157928               0.937249                1.979403   
299        HXT7   5.088683               2.129315                3.748323   
354     YOR186W   5.352607               0.532751                2.133530   
86       SPS100   6.607123               0.569632                2.592015   

    Gene Name 2  gene2_LFC  gene2_avg_exp_noNMPP1  gene2_avg_exp_yesNMPP1  
520     YOR214C   2.420921               0.249763                0.837610  
534      RPL20B  -4.111787               3.301923                2.202816  
543     YMR279C  -1.493332               0.720556                0.498756  
435      RPL41B  -1.805427               3.035567                2.703081  
417        TIR2   0.264392               1.615583                1.816379  
215        MIG2  -2.603948               1.995060                1.357922  
190        BAP3  -1.564176               2.917730                2.581011  
175     YBR016W   1.006656               2.482495                2.920191  
293       RSC30   0.186450               0.735845                0.823483  
31         YSR3   0.434324               1.766047                2.025983  
65         PMP2  -1.039738               1.206383                1.074730  
247       VID22  -0.182272               2.019608                2.089520  
397        AGA1   0.365022               2.392665                2.627318  
348        YRR1   1.239110               1.495062                1.991790  
416        AUS1   0.417175               1.621159                1.862177  
187        MRH1  -0.725184               2.009442                1.909999  
300        HXT1  -3.220434               3.064926                2.230077  
306        GCY1   5.895141               1.642249                3.551633  
330        BSC2   1.894483               1.292603                1.987821  
279     YLR046C   0.056061               2.117051                2.271243  
240     YLR346C   1.707717               0.637998                1.251560  
153        GPX1   4.660376               1.298613                2.826276  
537        DDR2   8.13

HXT1 and 6.  6 low expression - expressed a bit more.  1 high expression - decreased significantly. 
YPR1 and GCY1:  GCY1 highly expressed, YPR 1 induced a bit. 
HXT7/HXT4: 7 up, 4 down.  
YOR186W/YLR297W: 186W up 297W up a bit. 297W goes from nucleus to the vacuole underdna rep stress.  186W localized to vacuole.  
SPO19: one up one down, but not very highly expressed. Sporulation

both: 
GPX1/HYR1: GPX - oxidative stress, HYR1 oxidative stress sensor
HOR7/DDR2: DDR2 very highly expressed under stress.  function really not known (SGD)
PDR5/PDR15: ABC transporter
YBR284W/YJL070C: function not clear. 
SPS100/YGP1: SPS100 goes from barely expressed to high.  YGP1 already relatively highly expressed gets higher. Cell wall proteins.  SPS100 sporulation related. 

All jointly repressed: 
534	RPL20A	-3.936965	3.412258	2.367661	RPL20B	-4.111787	3.301923	2.202816
543	ATR1	-2.128023	2.374841	1.862896	YMR279C	-1.493332	0.720556	0.498756
435	RPL41A	-2.036833	3.657479	3.178033	RPL41B	-1.805427	3.035567	2.703081
215	MIG3	-1.339809	1.399376	1.173717	MIG2	-2.603948	1.995060	1.357922
190	BAP2	-1.192563	2.766405	2.572179	BAP3	-1.564176	2.917730	2.581011

Mig2/Mig3 cause glucose repression.  
ATR1 is a drug pump - SGD says it increases under stress.  Paralog might be a boron transporter- also should be induced (SGD)  but doesn't seem to be here. 
BAP2/3: amino acid uptake (really not decreased that much) 

In [141]:
#For SC genes that have no KL orthologs, what genes change drastically under PKA inhibition?  
#Already looked at paralogs above. 

sc_no_kl_orths = set(sc_avg_counts['sc_genename'])-set(kl_sc_PKA_data['sc_genename'])

sc_no_kl_orths_no_paralogs = sc_no_kl_orths - (set(ohnologs_no_kl_orth_pairs['Gene 1']) | set(ohnologs_no_kl_orth_pairs['Gene 2']))           
                                               
print("There are {} sc genes with no kl ortholog.  Of those {} do not have ohnologs".format(len(sc_no_kl_orths), len(sc_no_kl_orths_no_paralogs)))

columns_to_view = ['SC_common_name', 'log2FoldChange', 'AS_noNMPP1_YPD_t40_mean', 'AS_yesNMPP1_YPD_t40_mean']

sc_ypd_data_sc_only = sc_ypd_data.loc[sc_no_kl_orths]

sc_ypd_data_sc_only_pka_change = sc_ypd_data_sc_only[abs(sc_ypd_data_sc_only['log2FoldChange'])>2.0]

print("Of the {} sc genes with no kl ortholog, {} of them have fold change greater than 2.0 (up or down) under PKA inhibition.  20 of those are paralogs".format(
        len(sc_no_kl_orths), len(sc_ypd_data_sc_only_pka_change)))

sc_ypd_data_sc_only_pka_change.sort_values(by='log2FoldChange')[columns_to_view]



There are 1724 sc genes with no kl ortholog.  Of those 1668 do not have ohnologs
Of the 1724 sc genes with no kl ortholog, 336 of them have fold change greater than 2.0 (up or down) under PKA inhibition.  20 of those are paralogs


SC_common_name  log2FoldChange  AS_noNMPP1_YPD_t40_mean  \
sc_genename                                                           
YOR309C            YOR309C       -6.754461                 2.443750   
YGL168W               HUR1       -6.738577                 1.198944   
YBR092C               PHO3       -5.975478                 2.395164   
YDL039C               PRM7       -5.844456                 2.760269   
YDL227C                 HO       -5.497170                 2.999891   
YBR191W-A        YBR191W-A       -5.068049                 1.881674   
YDR281C               PHM6       -4.843624                 1.742183   
YPL130W              SPO19       -4.840507                 0.713965   
YMR042W              ARG80       -4.802050                 0.594620   
YOR302W            YOR302W       -4.696967                 0.606800   
YNR060W               FRE4       -4.695283                 2.218714   
YBR032W            YBR032W       -4.591691                 0.803248   
YDR183C-A        YDR183C-A       -4.570251                 1.984826   
YHR215W              PHO12       -4.499058                 2.588419   
YMR230W-A        YMR230W-A       -4.401609                 1.936622   
YOL014W            YOL014W       -4.122957                 1.869714   
YOR312C             RPL20B       -4.111787                 3.301923   
YEL045C            YEL045C       -4.035789                 0.522903   
YMR242C             RPL20A       -3.936965                 3.412258   
YBR069C               TAT1       -3.881612                 2.525078   
YMR304C-A        YMR304C-A       -3.876855                 2.583211   
YOR183W              FYV12       -3.794081                 0.664195   
YPL062W            YPL062W       -3.774677                 0.471529   
YBR296C              PHO89       -3.761152                 1.842630   
YKL131W            YKL131W       -3.757342                 0.449708   
YMR141C            YMR141C       -3.728189                 0.675079   
YBL053W            YBL053W       -3.619376                 0.446999   
YFR055W               IRC7       -3.618539                 1.551064   
YOR293C-A        YOR293C-A       -3.609671                 0.426937   
YFR032C-B        YFR032C-B       -3.581439                 0.996457   
...                    ...             ...                      ...   
YIR017W-A        YIR017W-A        4.588222                 0.077359   
YOL085C            YOL085C        4.589515                 0.286858   
YLR252W            YLR252W        4.602415                 1.455752   
YCR047W-A        YCR047W-A        4.635733                 0.077359   
YKL026C               GPX1        4.660376                 1.298613   
YFL013W-A        YFL013W-A        4.662115                 0.061998   
YGR087C               PDC6        4.701541                 0.412324   
YDR533C              HSP31        4.760134                 2.078395   
YIL054W            YIL054W        4.797116                 0.061998   
YER084W-A        YER084W-A        4.852067                 0.312180   
YFR052C-A        YFR052C-A        4.902003                 2.239221   
YDL024C               DIA3        4.953705                 0.280759   
YIL066W-A        YIL066W-A        4.985864                 0.061998   
YDR342C               HXT7        5.088683                 2.129315   
YMR090W            YMR090W        5.174739                 1.672798   
YGL156W               AMS1        5.221422                 1.112330   
YER121W            YER121W        5.279573                 0.599086   
YOR186W            YOR186W        5.352607                 0.532751   
YBL048W               RRT1        5.393732                 1.727007   
YER068C-A        YER068C-A        5.566594                 0.144496   
YOR120W               GCY1        5.895141                 1.642249   
YDR029W            YDR029W        5.943961                 0.000000   
YGL121C               GPG1        6.277594                 0.975415   
YMR174C               PAI3        6.

PAI3: Proteinase A inhibitor
GPG1: gamma subunit interacting with GPR1
RRT1: unknown function - deletion effect ribosomal dna transcription
AMS1: Oligosaccharide degradation.  
DIA3: pseudohyphal growth
HSP31: converts methylglyoxal to D-lactate;
PDC6: amino acid catabolism

Probably just overlapping HXK1:
YFR052C-A	YFR052C-A	4.902003	2.239221	3.865268
Probably just overlapping SYM1 (mitochondrial protein)
YLR252W	YLR252W	4.602415	1.455752	2.954681

Probably just overlapping PMR1
YGL168W	HUR1	-6.738577	1.198944	0.000000

# Subsets comparing K.Lac and S.Cer expression for differentially induced paralogs. 

Gene 1 Gene Name 1 kl_genename_gene1     Gene 2 Gene Name 2  \
0      YCL051W        LRE1      KLLA0C00803g    YDR528W        HLR1   
1      YCL050C        APA1      KLLA0C00847g    YDR530C        APA2   
2    YDR524C-B   YDR524C-B      KLLA0C00957g  YCL048W-A   YCL048W-A   
3      YCL048W       SPS22      KLLA0C01001g    YDR522C        SPS2   
4      YCL043C        PDI1      KLLA0C01111g    YDR518W        EUG1   
5      YCL040W        GLK1      KLLA0C01155g    YDR516C        EMI2   
6      YCL037C        SRO9      KLLA0C01210g    YDR515W        SLF1   
7      YCL036W        GFD2      KLLA0C01254g    YDR514C     YDR514C   
8      YCL035C        GRX1      KLLA0C01298g    YDR513W        GRX2   
9      YCL025C        AGP1      KLLA0C01606g    YDR508C        GNP1   
10     YCL024W        KCC4      KLLA0C01650g    YDR507C        GIN4   
11     YDR505C        PSP1      KLLA0C01716g    YLR177W     YLR177W   
12     YDR502C        SAM2      KLLA0C01782g    YLR180W        SAM1   
13     YDR501W        PLM2      KLLA0C01848g    YLR183C        TOS4   
14     YDR500C      RPL37B      KLLA0C01870g    YLR185W      RPL37A   
15     YLR187W        SKG3      KLLA0C01969g    YNL278W      CAF120   
16     YNL270C        ALP1      KLLA0C02343g    YEL063C        CAN1   
17     YHR155W        YSP1      KLLA0C02717g    YNL257C        SIP3   
18     YCR037C       PHO87      KLLA0C03454g    YJL198W       PHO90   
19     YCR034W        FEN1      KLLA0C03542g    YJL196C        ELO1   
20     YCR031C      RPS14A      KLLA0B07623g    YJL191W      RPS14B   
21     YJL181W     YJL181W      KLLA0A06237g    YJR030C     YJR030C   
22     YJL177W      RPL17B      KLLA0A06336g    YKL180W      RPL17A   
23     YJL165C        HAL5      KLLA0A06820g    YKL168C        KKQ8   
24     YJL164C        TPK1      KLLA0B07205g    YKL166C        TPK3   
25     YJL160C     YJL160C      KLLA0B07370g    YKL164C        PIR1   
26     YJL159W      HSP150      KLLA0B07392g    YKL163W        PIR3   
27     YJL139C        YUR1      KLLA0A05775g    YKR061W        KTR2   
28     YJL138C        TIF2      KLLA0A05731g    YKR059W        TIF1   
29     YJL137C        GLG2      KLLA0A05709g    YKR058W        GLG1   
..         ...         ...               ...        ...         ...   
489    YBR182C        SMP1      KLLA0E24025g    YPL089C        RLM1   
490    YBR181C       RPS6B      KLLA0E24047g    YPL090C       RPS6A   
491    YBR177C        EHT1      KLLA0E24201g    YPL095C        EEB1   
492    YBR172C        SMY2      KLLA0E24531g    YPL105C        SYH1   
493    YBR169C        SSE2      KLLA0E24597g    YPL106C        SSE1   
494    YPL112C       PEX25      KLLA0F00726g    YOR193W       PEX27   
495    YOR204W        DED1      KLLA0F01034g    YPL119C        DBP1   
496    YOR222W        ODC2      KLLA0D07073g    YPL134C        ODC1   
497    YOR226C        ISU2      KLLA0D07161g    YPL135W        ISU1   
498    YOR227W        HER1      KLLA0D07216g    YPL137C        GIP3   
499    YOR229W        WTM2      KLLA0D07282g    YPL139C        UME1   
500    YOR231W        MKK1      KLLA0D07304g    YPL140C        MKK2   
501    YOR233W        KIN4      KLLA0D07348g    YPL141C        FRK1   
502    YOR234C      RPL33B      KLLA0D07405g    YPL143W      RPL33A   
503    YOR237W        HES1      KLLA0D07480g    YPL145C        KES1   
504    YOR247W        SRL1      KLLA0C10054g    YPL163C        SVS1   
505    YOR256C        TRE2      KLLA0C09735g    YPL176C        TRE1   
506    YMR230W      RPS10B      KLLA0B08173g    YOR293W      RPS10A   
507    YMR233W        TRI1      KLLA0B13024g    YOR295W       UAF30   
508    YMR237W        BCH1      KLLA0F24981g    YOR299W        BUD7   
509    YMR243C        ZRC1      KLLA0F08723g    YOR316C        COT1   
510    YMR246W        FAA4      KLLA0B12936g    YOR317W        FAA1   
511    YML100W        TSL1      KLLA0E11265g    YMR261C        TPS3   
512    YMR264W        CUE1      KLLA0E11199g    YML101C        CUE4   
513    YML106W        URA5      

In [5]:
#build a dataframe that has all SC paralogs paired up with log fold change and mean expression columns.  Also has K.Lactis log fold change and mean expression columns. 

ohnologs_expression_gene1 = pd.merge(ohnologs, kl_sc_PKA_data_paralogs, how = 'inner', left_on = 'Gene 1', right_on = 'sc_genename')
ohnologs_expression = pd.merge(ohnologs_expression_gene1, kl_sc_PKA_data_paralogs, how = 'inner', left_on = 'Gene 2', right_on = 'sc_genename', suffixes = ['_gene1', '_gene2'])



#Assign sc threshold groups
high_threshold_SC = 2.0
low_threshold_SC = -high_threshold_SC
ohnologs_expression = ohnologs_expression.assign(threshold_group_SC = lambda x: io_library.threshold_group_SC_series(x.log2FoldChange_SC_gene1,x.log2FoldChange_SC_gene2,high_threshold_SC, low_threshold_SC))

#Assign kl threshold groups
high_threshold_KL = high_threshold_SC
low_threshold_KL = low_threshold_SC
ohnologs_expression = ohnologs_expression.assign(threshold_group_KL = lambda x: io_library.threshold_group_KL_series(x.log2FoldChange_KL_gene1, high_threshold_KL, low_threshold_KL))



#filter to genes with separated expression
foldchange_separation = 3.5

ohnologs_expression_separated = ohnologs_expression[abs(ohnologs_expression['log2FoldChange_SC_gene1'] - ohnologs_expression['log2FoldChange_SC_gene2']) > foldchange_separation]

columns_to_view = ['Gene 1', 'Gene Name 1', 'kl_genename_gene1', 'Gene 2', 'Gene Name 2', 'kl_genename_gene2', 'log2FoldChange_SC_gene1', 'log2FoldChange_SC_gene2', 'threshold_group_SC','log2FoldChange_KL_gene1','threshold_group_KL']

ohnologs_expression_separated[columns_to_view]




Gene 1 Gene Name 1 kl_genename_gene1     Gene 2 Gene Name 2  \
0      YCL051W        LRE1      KLLA0C00803g    YDR528W        HLR1   
2    YDR524C-B   YDR524C-B      KLLA0C00957g  YCL048W-A   YCL048W-A   
6      YCL037C        SRO9      KLLA0C01210g    YDR515W        SLF1   
19     YCR034W        FEN1      KLLA0C03542g    YJL196C        ELO1   
24     YJL164C        TPK1      KLLA0B07205g    YKL166C        TPK3   
26     YJL159W      HSP150      KLLA0B07392g    YKL163W        PIR3   
35     YJL110C        GZF3      KLLA0F21296g    YKR034W       DAL80   
42     YJL079C        PRY1      KLLA0D02420g    YKR013W        PRY2   
50     YFR017C        IGD1      KLLA0A10967g    YOL024W     YOL024W   
56     YLR249W        YEF3      KLLA0F12210g    YNL014W        HEF3   
60     YCR005C        CIT2      KLLA0F12760g    YNR001C        CIT1   
66     YEL039C        CYC7      KLLA0F16929g    YJR048W        CYC1   
69     YJR054W     YJR054W      KLLA0A08063g    YML047C        PRM6   
92     YNL104C        LEU4      KLLA0F23529g    YOR108W        LEU9   
96     YNL093W       YPT53      KLLA0C13728g    YOR089C       VPS21   
97     YNL087W        TCB2      KLLA0C13860g    YOR086C        TCB1   
106    YIL113W        SDP1      KLLA0F03597g    YNL053W        MSG5   
114    YDL085W        NDE2      KLLA0E21891g    YMR145C        NDE1   
119    YLR029C      RPL15A      KLLA0F17633g    YMR121C      RPL15B   
122    YKL141W        SDH3      KLLA0E12123g    YMR118C        SHH3   
126    YKL127W        PGM1      KLLA0B12694g    YMR105C        PGM2   
132    YKL091C     YKL091C      KLLA0B06479g    YMR079W       SEC14   
146    YKL072W        STB6      KLLA0F26499g    YMR053C        STB2   
149    YNL309W        STB1      KLLA0E07283g    YOL131W     YOL131W   
160    YBR147W        RTC2      KLLA0C03762g    YOL092W     YOL092W   
167    YDL022W        GPD1      KLLA0C04004g    YOL059W        GPD2   
168    YDL021W        GPM2      KLLA0C04081g    YOL056W        GPM3   
176    YBR042C       CST26      KLLA0B06820g    YDR018C     YDR018C   
181  YDR034W-B   YDR034W-B      KLLA0F18766g  YBR056W-A   YBR056W-A   
188    YBR104W        YMC2      KLLA0B08503g    YPR058W        YMC1   
..         ...         ...               ...        ...         ...   
331    YLR273C        PIG1      KLLA0D09240g    YOR178C        GAC1   
334    YLR293C        GSP1      KLLA0C05126g    YOR185C        GSP2   
337    YBR214W       SDS24      KLLA0C05588g    YGL056C       SDS23   
357    YMR177W        MMT1      KLLA0C16181g    YPL224C        MMT2   
360    YMR182C        RGM1      KLLA0C16005g    YPL230W        USV1   
376  YNR034W-A   YNR034W-A      KLLA0F27390g  YCR075W-A   YCR075W-A   
379    YAL062W        GDH3      KLLA0F00594g    YOR375C        GDH1   
385    YAL034C       FUN19      KLLA0F02365g    YOR338W     YOR338W   
398    YER054C        GIP2      KLLA0F27533g    YIL045W        PIG2   
402    YER062C        HOR2      KLLA0C08217g    YIL053W        RHR2   
405    YER070W        RNR1      KLLA0C07887g    YIL066C        RNR3   
408    YIL079C        AIR1      KLLA0F05896g    YDL175C        AIR2   
419    YBL054W        TOD6      KLLA0E19933g    YER088C        DOT6   
422    YBL061C        SKT5      KLLA0E20263g    YER096W        SHC1   
424    YBL068W        PRS4      KLLA0E20373g    YER099C        PRS2   
437    YBL039C        URA7      KLLA0E05545g    YJR103W        URA8   
438    YBL043W       ECM13      KLLA0E05677g    YJR115W     YJR115W   
442    YFL048C       EMP47      KLLA0D05863g    YLR080W       EMP46   
454    YDR077W        SED1      KLLA0E01057g    YER150W        SPI1   
460    YDR107C        TMN2      KLLA0F18931g    YLR083C       EMP70   
470    YDR178W        SDH4      KLLA0D18117g    YLR164W        SHH4   
475    YDR222W     YDR222W      KLLA0F13200g    YLR225C     YLR225C   
476    YDR223W        CRF1      KLLA0F13222g    YLR223C        IFH1   
477    YDL204W        RTN2      KLLA0C15081g    YDR233C        RTN1   
493    YBR169C        SSE2      

In [270]:
#Bar chart showing expression groups in separated and not separated sets. 

expression_groups = ['flat_flat', 'up_flat','down_down','down_flat', 'up_up','up_down']

data_sets = {'All Paralogs': ohnologs_expression, 'Separated Paralogs': ohnologs_expression_separated}

# data_sets['Separated Paralogs']['threshold_group'].value_counts()[expression_groups]

traces = []

for name, data_set in data_sets.items():

    counts = data_set['threshold_group_SC'].value_counts()[expression_groups].fillna(0)
    total_counts = sum(counts)
    percentages = counts/total_counts*100
    
    trace = pygo.Bar(
        x=expression_groups,
        y=percentages,
        text=['{} <br> {:2.0f}%'.format(item, item/total_counts*100) for item in list(counts)],
        textposition = 'auto',
        insidetextfont = {'color': 'white'},
        name = name
    #     marker=dict(
    #             color='rgb(158,202,225)',
    #             line=dict(
    #                 color='rgb(8,48,107)',
    #                 width=1.5),
    #             ),
    #     opacity=0.6
        )
    
    traces.append(trace)

data = traces
layout = pygo.Layout(
    barmode='group'
)

fig = pygo.Figure(data=data, layout=layout)
py.iplot(fig, filename='paralog_expression_groups')




In [281]:
#Bar chart showing KL expressionin separated genes by group

expression_groups = ['flat_flat', 'up_flat','down_down','down_flat', 'up_up','up_down']
kl_foldchange_groups = ['up', 'flat', 'down']

#build dataframe with counts for each level

data_set = ohnologs_expression #ohnologs_expression_separated

count_data = {}
for group in kl_foldchange_groups: 

    kl_group = data_set[data_set['threshold_group_KL']==group]

    counts = kl_group['threshold_group_SC'].value_counts()[expression_groups].fillna(0)

    count_data[group] = counts

count_data_df = pd.DataFrame(count_data, index = expression_groups)

traces = []

for group in kl_foldchange_groups:

    counts = count_data_df[group]
    #total_counts = sum(counts)
    #percentages = counts/total_counts*100
    
    trace = pygo.Bar(
        x=expression_groups,
        y=counts,
        text= counts,
        textposition = 'auto',
        insidetextfont = {'color': 'white'},
        name = 'kl exp ' + group
    #     marker=dict(
    #             color='rgb(158,202,225)',
    #             line=dict(
    #                 color='rgb(8,48,107)',
    #                 width=1.5),
    #             ),
    #     opacity=0.6
        )
    
    traces.append(trace)

data = traces
layout = pygo.Layout(
    barmode='stack',
    xaxis = {'title': 'SCer paralog expression groups'},
    yaxis = {'title': 'Number of paralog pairs'}
)

fig = pygo.Figure(data=data, layout=layout)
py.iplot(fig, filename='paralog_expression_groups_kl_exp')


In [197]:
AB

A    B
0 -2.5  2.2
1  1.0  4.1

In [203]:




df = pd.DataFrame([[-2.5,2.2], [1.0,4.1]], columns =['A','B'] )
df

#io_library.threshold_group_series(A,B, high_threshold, low_threshold)

# io_library.threshold_group(b,a,high_threshold, low_threshold)

# #input: a, b, high_threshold, low_threshold
# #output: group - either {'up_up', 'up_flat','up_down','flat_flat', 'down_flat','down_down'}

# a_sign = io_library.threshold_sign(a, high_threshold, low_threshold)
# b_sign = io_library.threshold_sign(b, high_threshold, low_threshold)

# sign_sum = a_sign+b_sign

# if sign_sum ==2:
#     group = 'up_up'
# elif sign_sum == -2: 
#     group = 'down_down'
# elif sign_sum == 1: 
#     group = 'up_flat'
# elif sign_sum == -1: 
#     group = 'down_flat'
# elif sign_sum == 0:
#     if a_sign-b_sign == 0:
#         group = 'flat_flat'
#     else: 
#         group = 'up_down'
# else: 
#     print('Error - did not assign group')

# return group



A    B      new
0 -2.5  2.2  up_down
1  1.0  4.1  up_flat

In [ ]:
#bar chart

x = ['Product A', 'Product B', 'Product C']
y = [20, 14, 23]
y2 = [16,12,27]

trace1 = go.Bar(
    x=x,
    y=y,
    text=y,
    textposition = 'auto',
    marker=dict(
        color='rgb(158,202,225)',
        line=dict(
            color='rgb(8,48,107)',
            width=1.5),
        ),
    opacity=0.6
)

trace2 = go.Bar(
    x=x,
    y=y2,
    text=y2,
    textposition = 'auto',
    marker=dict(
        color='rgb(58,200,225)',
        line=dict(
            color='rgb(8,48,107)',
            width=1.5),
        ),
    opacity=0.6
)

data = [trace1,trace2]

py.iplot(data, filename='grouped-bar-direct-labels')




In [19]:
ohnologs_expression[ohnologs_expression['Gene Name 1']=='ADH5'][columns_to_view]



Gene 1 Gene Name 1 kl_genename_gene1   Gene 2 Gene Name 2  \
163  YBR145W        ADH5      KLLA0F21010g  YOL086C        ADH1   

    kl_genename_gene2  log2FoldChange_SC_gene1  log2FoldChange_SC_gene2  \
163      KLLA0F21010g                 0.148592                 0.099784   

    threshold_group_SC  log2FoldChange_KL_gene1 threshold_group_KL  
163          flat_flat                 0.096766               flat